source/core/ib_stencils.cc

```diff
@@ -0,0 +1,272 @@
+        interpolation_points[2]=interpolation_point_3;
+
+        return {dof_point,interpolation_points};
+    }
+    if (order==4) {
+        Tensor<1, dim, double> vect_ib =(dof_point - p.position -p.radius *(dof_point-p.position)/(dof_point-p.position).norm());
+
+        Point<dim, double> interpolation_point_1(dof_point+vect_ib* 1./32.);
+
+        Point<dim, double> interpolation_point_2(dof_point+vect_ib* 1/16.);
+
+        Point<dim, double> interpolation_point_3(dof_point+vect_ib* 3./32.);
+
+        Point<dim, double> interpolation_point_4(dof_point+vect_ib* 1./8.);
+
+        std::vector<Point<dim>> interpolation_points(4);
+        interpolation_points[0]=interpolation_point_1;
+        interpolation_points[1]=interpolation_point_2;
+        interpolation_points[2]=interpolation_point_3;
+        interpolation_points[3]=interpolation_point_4;
+
+        return {dof_point,interpolation_points};
+    }
+    if (order>4) {
+        // In this case the cell is directly used to find the solution at the IB position. In this case only one point is needed (the position of the point on the IB).
+        Tensor<1, dim, double> vect_ib =(dof_point - p.position -p.radius *(dof_point-p.position)/(dof_point-p.position).norm());
+
+        Point<dim, double> ib_point(dof_point - vect_ib);
+
+        Point<dim, double> interpolation_point_1(dof_point+vect_ib* 1./8.);
+
+        std::vector<Point<dim>> interpolation_points(1);
+        interpolation_points[0]=interpolation_point_1;
+
+        return {ib_point,interpolation_points};
+    }
+
+    Tensor<1, dim, double> vect_ib = (dof_point - p.position -
+                                      p.radius * (dof_point - p.position) / (dof_point - p.position).norm());
+
+
+    Point<dim, double> ib_point(dof_point - vect_ib);
+
+    Point<dim, double> interpolation_point_1(dof_point + vect_ib * 1 / 8);
+
+    std::vector<Point < dim>>interpolation_points(1);
+    interpolation_points[0]=interpolation_point_1;
+
+    return {ib_point,interpolation_points};
+}
+
+
+template<int dim>
+double
+IBStencils<dim>::vitesse_ib(IBParticle<dim> p,Point<dim> dof_point,unsigned int component) {
+    // Return the value of the IB condition for that specific stencil.
+    double v_ib=0;
+    if (dim == 2) {
+        if (component == 0) {
+            //vx in 2D
+            v_ib = -p.omega[2] *
+                 p.radius *
+                 ((dof_point -
+                  p.position) /
+                 (dof_point -
+                  p.position)
+                         .norm())[1] +
+                    p.velocity[0];
+        }
+        if (component == 1) {
+            //vy in 2D
+            v_ib = p.omega[2] *
+                 p.radius *
+                 ((dof_point -
+                   p.position) /
+                  (dof_point -
+                   p.position)
+                          .norm())[0] +
+                 p.velocity[1];
+
+        }
+    }
+    if (dim == 3) {
```

Comment:
Did you apply automatic indentation here ? That look sa bit messed up. Also, if you use a cross product,  why not use the cross product functionnality of deal.II? It's already there.

Reply:
apply both of those recommendation